src/main/java/dev/galacticraft/mod/events/GCEventHandlers.java

```diff
@@ -75,6 +82,48 @@ public static void onWorldTick(ServerLevel level) {
         level.galacticraft$getSealerManager().tick();
     }
 
+    public static void onServerTick(MinecraftServer server) {
+        // throw meteors around players
+        PlayerLookup.all(server).forEach(player -> {
+            ServerLevel level = player.serverLevel();
+            Holder<CelestialBody<?, ?>> celestialBody = level.galacticraft$getCelestialBody();
+            if (celestialBody == null) return;
+
+            // calculate frequency of meteors on current celestial body
+            float atmospherePressure = celestialBody.value().atmosphere().pressure();
+            float frequency = (atmospherePressure <= Mth.EPSILON) ? 5.0f : (atmospherePressure * 100.0f);
+
+            // throw meteor
+            int chance = (int) (frequency);
+            if (level.random.nextInt(chance) == 0) {
+                throwMeteor(server, level, player, 1);
+            }
+
+            // throw bigger meteor if you're lucky enough
+            if (level.random.nextInt(chance * 3) == 0) {
+                throwMeteor(server, level, player, 6);
+            }
+        });
+    }
+    private static void throwMeteor(MinecraftServer server, ServerLevel level, Player targetPlayer, int meteorSize) {
+        Player nearestPlayer = level.getNearestPlayer(targetPlayer, 100.0);
+        if (nearestPlayer == null || nearestPlayer.getId() > targetPlayer.getId()) return;
+
+        int maxOffset = server.getPlayerList().getViewDistance() * Constant.Chunk.WIDTH - 1;
+
+        int offsetX = Math.min(maxOffset, level.random.nextInt(20) + 160);
+        int offsetZ = level.random.nextInt(20) - 10;
+        double deltaX = level.random.nextDouble() * 2.0 - 2.5;
+        double deltaZ = level.random.nextDouble() * 5.0 - 2.5;
```

Comment:
Is `deltaX` meant to be skewed towards negative values? The range of values `deltaX` can take is -4.5 to -0.5, since `nextDouble()` is between 0 and 1, which just seems a bit strange (although maybe there is a good reason I don't know about).

Doing something like this might be more intuitive:

```suggestion
        double deltaX = (level.random.nextDouble() - 0.5) * 2.0;
        double deltaZ = (level.random.nextDouble() - 0.5) * 5.0;
```

Reply:
I have realised why `deltaX` is skewed towards negative values, it is because `offsetX` is always positive, so it sort of balances out. This system is a bit flawed, as the meteors can only come from the East and the distribution is not ideal. It might be best to revert this back to the previous expression until this can be redesigned properly, unless you want to have a go at that now.

Sorry for the confusion.